### ISSUE TITLE
Error repeated in log for SkyQ Mini

### ISSUE THREAD

**Issue by ollo69** [RogerSelwyn/Home_Assistant_SkyQ_MediaPlayer#4]
**Describe the bug**

Hi, 

first of all thanks for this fantastic component, very useful for me.
Today I found a lot of error in the log related to the SkyQ mini connected to a SkyQ platinum (italian version)
Both SkyQ Platinum and SkyQ Mini was turned-on.
Error log start about at 9.00am and continue until I reboot Home Assistant (see log below).

**To Reproduce**
1. What were you doing/watching at the time ? Live TV
2. What channel were you on? I think Sky TG24
3. What time did the problem occur? see log
4. Where are you (bear in mind component supports GBR/ITA at the moment)? ITA

**Your config.yaml**
```yaml
media_player:
 - platform:  skyq
   name: SkyQ Mini Letto
   host: x.x.x.x
   live_tv: True
   country: "IT"
````
**Output of HA logs**

Paste the relavant output of the HA log here.

```
Update for media_player.skyq_mini_letto fails
12:32:08 – custom_components/skyq/media_player.py (ERROR) - il messaggio si è verificato per la prima volta alle 8:57:21 e compare 1176 volte
 
Dettagli registro (ERROR)
Logger: homeassistant.helpers.entity
Source: custom_components/skyq/media_player.py:388
First occurred: 8:57:21 (1176 occurrences)
Last logged: 12:32:08

Update for media_player.skyq_mini_letto fails
Traceback (most recent call last):
  File "/home/homeassistant/.homeassistant/custom_components/skyq/media_player.py", line 357, in _async_getCurrentMedia
    self._remote.getCurrentMedia
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/srv/homeassistant/lib/python3.7/site-packages/pyskyqremote/skyq_remote.py", line 169, in getCurrentMedia
    channel = self._getChannelNode(sid)["channel"]
  File "/srv/homeassistant/lib/python3.7/site-packages/pyskyqremote/skyq_remote.py", line 471, in _getChannelNode
    s for s in self._channels["services"] if s["sid"] == str(sid)
KeyError: 'services'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/helpers/entity.py", line 470, in async_device_update
    await self.async_update()
  File "/home/homeassistant/.homeassistant/custom_components/skyq/media_player.py", line 268, in async_update
    await self._async_updateCurrentProgramme()
  File "/home/homeassistant/.homeassistant/custom_components/skyq/media_player.py", line 342, in _async_updateCurrentProgramme
    await self._async_getCurrentMedia()
  File "/home/homeassistant/.homeassistant/custom_components/skyq/media_player.py", line 388, in _async_getCurrentMedia
    f"X0020M - Current Media retrieval failed: {currentMedia} : {err}"
UnboundLocalError: local variable 'currentMedia' referenced before assignment
Connessione persa. Riconnessione…

```

**Comment by RogerSelwyn** [comment]
And I should probably add as a question. What version are you using. I would note that you shouldn't be using country IT any more. If you have an Italian box, then you don't need to have anything. Or if you have another country which uses Italian EPG, then you need ITA.

**Comment by ollo69** [comment]
Sorry, I wrote IT but in config.yaml is set to "ITA". My box is Italian
Version of HA is 0.109.6.

**Comment by RogerSelwyn** [comment]
Not, what version of the component are you using....

**Comment by ollo69** [comment]
Oh, ok. Last public release (v2.1.7)

**Comment by RogerSelwyn** [comment]
When I look at that error trace, the line numbers don't align to v2.1.17. Can you either re-install v2.1.17 and reboot or upgrade to the latest beta. Recognising this is an intermittent issue, let me know if it occurs again.

**Comment by RogerSelwyn** [comment]
Just one thought though. Had your HA instance been running for some time or was it after a reboot?

**Comment by ollo69** [comment]
Was running from some time (at list 12 hours)

**Comment by RogerSelwyn** [comment]
12 hours isn't long. The initiating error suggests that it had lost the list of channels to work against. Or it maybe it was a new channel number, but I'm not sure enough on the error message. In any case, let me know if it occurs again.

**Comment by ollo69** [comment]
What was surprising me was not the error but the repetition of it. Probably an error create some incoherent situation that the component was not able to recover and the only solution was the reboot of HA.

**Comment by RogerSelwyn** [comment]
This is a polling component, so it checks SkyQ box every 10 seconds. So you will get errors every 10 seconds if it blows up.

**Comment by ollo69** [comment]
For this reason  error should be recovered and not require reboot.
I will inform you if it will occur again, trying to eventually understand the root cause.

**Comment by RogerSelwyn** [comment]
I don't understand how the channel list was lost, but I have added a change to re-retrieve. This will also re-retrieve in the event the channel list has been changed by Sky.

**Comment by RogerSelwyn** [comment]
This issue should be fixed by v2.1.8. If it re-occurs please let me know.

https://github.com/RogerSelwyn/Home_Assistant_SkyQ_MediaPlayer/releases/tag/v2.1.8
